docs/Integrating Your HMI/Vehicle Data/index.md

```diff
@@ -0,0 +1,169 @@
+                    "name": "customStructVal",
+                    "key": "KEY_CUSTOM_STRUCT_VAL",
+                    "type": "String",
+                    "mandatory": true
+                },
+                {
+                    "name": "customStructVal2",
+                    "key": "KEY_CUSTOM_STRUCT_VAL2",
+                    "minvalue": 0,
+                    "maxvalue": 100,
+                    "type": "Integer",
+                    "mandatory": true
+                },
+                {
+                    "name": "customDeprecatedVal",
+                    "key": "KEY_CUSTOM_DEPRECATED_VAL",
+                    "minvalue": 0,
+                    "maxvalue": 100,
+                    "type": "Integer",
+                    "mandatory": true,
+                    "until": "7.0"
+                },
+                {
+                    "name": "customDeprecatedVal",
+                    "key": "KEY_CUSTOM_DEPRECATED_VAL",
+                    "minvalue": 0,
+                    "maxvalue": 100,
+                    "type": "Integer",
+                    "mandatory": true,
+                    "deprecated": true
+                    "since": "7.0"
+                }
+            ],
+            "key": "KEY_CUSTOM_STRUCT",
+            "type": "Struct",
+            "mandatory": false
+        }
+    ]
+}
+```
+
+In addition to custom items, this feature can be used to expose other vehicle data items that were introduced to the project in later versions. This can be useful when the software version on the head unit cannot be updated easily. If a vehicle data item is added into the project, the definition of this item will be included in the policy table by default. Any vehicle data items which are defined in Core's local Mobile API will be ignored from the policy table, but newer items will be interpreted as custom items. This allows apps to use these data items normally if they are exposed by the head unit, even when they were not initially supported.
+
+### Custom Vehicle Data Requests
+
+Custom vehicle data requests have a separate structure to normal vehicle data requests. While normal vehicle data items are requested using the key structure of `"<item.name>: true"`, custom items are constructed using the `item.key` field and can have a nested structure (when requesting `Struct` items). For example, when requesting all of the vehicle data items which are defined above, the HMI would receive the following message:
+
+```
+{
+    "id" : 139,
+    "jsonrpc" : "2.0",
+    "method" : "VehicleInfo.GetVehicleData",
+    "params" : {
+        "KEY_CUSTOM_STRING": true,
+        "KEY_CUSTOM_INT": true,
+        "KEY_CUSTOM_FLOAT": true,
+        "KEY_CUSTOM_BOOL": true,
+        "KEY_CUSTOM_ARRAY": true,
+        "KEY_CUSTOM_STRUCT": {
+            "KEY_CUSTOM_STRUCT_VAL": true,
+            "KEY_CUSTOM_STRUCT_VAL2": true,
+            "KEY_CUSTOM_DEPRECATED_VAL": true
+        }
+    }
+}
+```
+
+### Vehicle Data Mapping File
```

Comment:
Do we have any example mapping files available? Maybe in ATF?

Reply:
While there is technically one example of this in ATF, it does not represent the only way that this can be structured, so I'm not certain how useful it will be. I can include it if you would like.

```json
{
   "version":"0.0.1",
   "date":"01-01-2020",
   "vehicleDataTable":[
      {
         "CGEA1.3c":{
            "defaultPowertrain":{
               "vehicleData":[

               ]
            },
            "PHEV":{
               "vehicleData":[
                  {
                     "key":"OEM_REF_FUELLEVEL",
                     "type":"Integer",
                     "minFrequency":200,
                     "maxLatency":10,
                     "messageName":"Cluster_Info3",
                     "messageID":"0x434",
                     "signalName":"FuelLvl_Pc_Dsply",
                     "transportChannel":"HS3",
                     "resolution":0.109,
                     "offset":-5.2174
                  }
               ]
            }
         }
      }
   ]
}
```

(modified from https://github.com/smartdevicelink/sdl_atf_test_scripts/blob/master/files/jsons/GenericNetworkSignalData/OEM_Mapping_update.json)

Reply:
OK good point, and I dont find this to be helpful at all as an example anyways /shrug